### PR TITLE
Defect/434 null results exception not wrapped

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/PipelineTestResult.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 import org.opentestsystem.rdw.script.ScriptError;
 
+import java.util.Collections;
```
Comment (burzum619) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/model/PipelineTestResult.java]: unused import maybe?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultPipelineService.java [hunk 2]
```diff
@@ -210,15 +210,16 @@ private PipelineTestRun run(final PipelineTest test, final List<PipelineScript>
...
-                    error
-            )))
-                    .build();
+            return run.result(PipelineTestResult.failedWithScriptError(
+                    new ScriptError(
+                            exception.getMessage(),
+                            exception.getLineNumber()
+                    )
+            )).build();
+        } catch (final RuntimeException exception) {
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultPipelineService.java]: @gregcharles should it be the pipeline's job to wrap these as `ScriptRuntimeExceptions` or should i be catching them here? wasn't sure so just did this.

File: admin-service/src/main/resources/pipeline-script-samples/csv.groovy [hunk 3]
```diff
@@ -1 +1,7 @@
-// csv sample script goes here
+// enables csv utilities
+enable 'csv'
+
+// add your transformations here
+
+// outputs transformed csv
+outputCsv
```
Comment (esotrope) [inline admin-service/src/main/resources/pipeline-script-samples/csv.groovy]: better csv sample - thanks @gregcharles

File: webapp/src/main/webapp/src/app/admin/ingest-pipeline/page/pipeline/pipeline.component.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ const defaultCompileDebounceTime = 2000;
 function compilationErrorToMessage(value: ScriptError): Message {
   return {
     type: <MessageType>'error',
-    row: value.row != null ? value.row - 1 : undefined,
+    row: value.row != null ? value.row - 1 : 0,
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/ingest-pipeline/page/pipeline/pipeline.component.ts]: default to show errors without rows on the first row

File: webapp/src/main/webapp/src/app/admin/ingest-pipeline/page/pipelines/pipelines.component.html [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+    <ng-container *ngFor="let pipeline of _pipelines">
+      <pipeline-card
+        [pipeline]="pipeline"
+        [routerLink]="[pipeline.id]"
+      ></pipeline-card>
+    </ng-container>
+  </div>
+</ng-container>
+<ng-template #loading>
+  <spinner></spinner>
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/ingest-pipeline/page/pipelines/pipelines.component.html]: added loading spinner so the page isn't just blank for a second

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/PipelineTestResult.java (hunk 1 above)
Comment (esotrope) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/model/PipelineTestResult.java]: sure is : j

**Pull request merged**